File: 122.买卖股票的最佳时机-ii.py
#
# @lc app=leetcode.cn id=122 lang=python3
#
# [122] 买卖股票的最佳时机 II
#

# @lc code=start
import numpy
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        # 等价于求所有上升区间的和
        result = 0
        for i in range(1,len(prices)):
            result += max(prices[i] - prices[i-1],0)
        return result
        
# @lc code=end

File: 122.买卖股票的最佳时机-ii.cpp
/*
 * @lc app=leetcode.cn id=122 lang=cpp
 *
 * [122] 买卖股票的最佳时机 II
 */
#include <iostream>
#include <vector>
using namespace std;
// @lc code=start
class Solution
{
public:
    int maxProfit(vector<int> &prices)
    {
        // 等价于所有上升区间的和
        int result = 0;
        for (int i = 1; i < prices.size(); i++)
        {
            if (prices[i] > prices[i - 1])
            {
                result += prices[i] - prices[i - 1];
            }
        }
        return result;
    }
};
// @lc code=end
